tools/ab_test.py

```diff
@@ -180,7 +180,7 @@ def uninteresting_dimensions(processed_emf):
     return uninteresting
 
 
-def collect_data(binary_dir: Path, pytest_opts: str):
+def collect_data(tag: str, binary_dir: Path, pytest_opts: str):
```

Comment:
iirc, the location of the json files can be modified via pytest's `--json-report-file` option. I think it'll be cleaner to use this, and modify our test framework to dynamically determine `TEST_ARTIFACTS_DIR` to be the parent of the file passed via --json-report-file.